core/templates/pages/exploration-editor-page/modal-templates/exploration-modify-translations-modal.component.ts

```diff
@@ -0,0 +1,116 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for updating translations of edited content.
+ */
+
+import {Component, Input} from '@angular/core';
+import {NgbActiveModal} from '@ng-bootstrap/ng-bootstrap';
+import {ConfirmOrCancelModal} from 'components/common-layout-directives/common-elements/confirm-or-cancel-modal.component';
+import {ContextService} from 'services/context.service';
+import {EntityBulkTranslationsBackendApiService} from '../services/entity-bulk-translations-backend-api.service';
+import {EntityTranslationsService} from 'services/entity-translations.services';
+import {LanguageUtilService} from 'domain/utilities/language-util.service';
+import {TranslatedContent} from 'domain/exploration/TranslatedContentObjectFactory';
+import {ChangeListService} from '../services/change-list.service';
+
+interface LanguageCodeToContentTranslations {
+  [language_code: string]: TranslatedContent;
+}
+
+@Component({
+  selector: 'oppia-exploration-modify-translations-modal',
+  templateUrl: './exploration-modify-translations-modal.component.html',
+})
+export class ModifyTranslationsModalComponent extends ConfirmOrCancelModal {
+  @Input() contentId!: string;
+  explorationId!: string;
+  explorationVersion!: number;
+  contentTranslations: LanguageCodeToContentTranslations = {};
+  allExistingTranslationsHaveBeenRemoved: boolean = false;
+
+  constructor(
+    private ngbActiveModal: NgbActiveModal,
+    private contextService: ContextService,
+    private entityBulkTranslationsBackendApiService: EntityBulkTranslationsBackendApiService,
+    private languageUtilService: LanguageUtilService,
+    private entityTranslationsService: EntityTranslationsService,
+    private changeListService: ChangeListService
+  ) {
+    super(ngbActiveModal);
+  }
+
+  ngOnInit(): void {
+    this.explorationId = this.contextService.getExplorationId();
+    this.explorationVersion =
+      this.contextService.getExplorationVersion() as number;
+
+    // Populate the content translations via latest draft changes first,
+    // in order to get the most recently updated translations.
+    for (let language in this.entityTranslationsService
+      .languageCodeToEntityTranslations) {
+      let translationContent =
+        this.entityTranslationsService.languageCodeToEntityTranslations[
+          language
+        ].getWrittenTranslation(this.contentId);
+      if (translationContent) {
+        this.contentTranslations[language] = translationContent;
+      }
+    }
+
+    this.changeListService.getTranslationChangeList().forEach(changeDict => {
```

Comment:
Just to check, do we need to iterate over the entire list here to set the `allExistingTranslationHaveBeenRemoved` boolean or can we exit the loop after first case matching the condition is found? If it is the latter, it might be better to use the "[some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)" array operation.

Reply:
We don't need to iterate over the entire list. We could use a normal for loop and just break when it's set to true, does that work?

Reply:
The "some" array has this baked in (see documentation linked in my previous comment.)

Reply:
Done!